test/support/generator/random-value-generator.ts

```diff
@@ -29,10 +29,15 @@ export class RandomValueGenerator {
     return this.chance.string({ length, pool: '0123456789' });
   }
 
+  // TODO apim-474 This has the potential to be flakey. Realistically - We should never get so unlucky as to trigger the loop
```

Comment:
I could avoid this by reducing the letter pool down a touch, as we already are testing this regex elsewhere. Will do that shortly